File: trunk/ShadowFrameworkLite20/src/shadow/system/data/SFAlternativeDataCenter.java
package shadow.system.data;


public abstract class SFAlternativeDataCenter implements SFIDataCenter{

	protected SFIDataCenter dataCenter = null;

	public SFAlternativeDataCenter() {
		super();
	}

	public void setup() {
		dataCenter=SFDataCenter.getDataCenter().getDataCenterImplementation();
		SFDataCenter.setDataCenterImplementation(this);
	}

	public void unset() {
		if(dataCenter!=null){
			SFDataCenter.setDataCenterImplementation(dataCenter);
		}
	}

}

File: trunk/ShadowFrameworkJS/src/shadow/system/data/SFAlternativeDataCenter.js




function SFAlternativeDataCenter(){
	this.dataCenter=null;
}

SFAlternativeDataCenter.prototype["setup"]=function(){
	this.dataCenter=SFDataCenter_getDataCenter().getDataCenterImplementation();
	SFDataCenter_setDataCenterImplementation(this);
};
	
SFAlternativeDataCenter.prototype["unset"]=function(){
	if(this.dataCenter!=null){
		SFDataCenter_setDataCenterImplementation(this.dataCenter);
	}
	this.dataCenter=null;
};
	
